crates/ruff_linter/src/rules/pyflakes/rules/unused_import.rs

```diff
@@ -288,30 +314,31 @@ pub(crate) fn unused_import(checker: &Checker, scope: &Scope, diagnostics: &mut
                         first_party: true,
                         ..
                     })
-                )
+                ) && preview_mode
             });
 
         // generate fixes that are shared across bindings in the statement
-        let (fix_remove, fix_reexport) = if (!in_init || fix_init) && !in_except_handler {
```

Comment:
This section looks like a large change but isn't. Turn on `-w` to see the diff w/o whitespace changes.